### PR TITLE
fix(api): active update

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What changed? Why was the change needed?
* Workflow `active` is respected when `false`

### Screenshots
_Demo of active toggling in Dashboard_

https://github.com/user-attachments/assets/a05d2e8d-869a-4fd8-bab7-2dee6dfa7061

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export class UpdateWorkflow {
       updatePayload.name = command.name;
     }
 
-    if (command.active) {
+    if (command.active !== undefined && command.active !== null) {
```
Comment (djabarovgeorge) [inline libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts]: looks like it was broken for a while

File: apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts [hunk 2]
```diff
@@ -184,6 +183,11 @@ export class UpsertWorkflowUseCase {
     );
   }
 
+  /**
+   * Upsert workflow preferences. While this operation is not typically needed
+   * in a standard workflow update, it's maintained here to support environment
+   * sync scenarios and to enable code reusability.
+   */
```
Comment (djabarovgeorge) [inline apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts]: good comment to have :)

**Review by SokratisVidros — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

File: apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts [hunk 3]
```diff
@@ -76,7 +76,6 @@ export class UpsertWorkflowUseCase {
   ) {}
   async execute(command: UpsertWorkflowCommand): Promise<WorkflowResponseDto> {
     const workflowForUpdate = await this.queryWorkflow(command);
-
     const workflow = await this.createOrUpdateWorkflow(workflowForUpdate, command);
```
Comment (rifont) [inline apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts:79]:
```suggestion

```

File: apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts (hunk 2 above)
Comment (rifont) [reply apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts]: Removing the comment as prefs are removed from this use-case in https://github.com/novuhq/novu/pull/7005, avoiding a conflict now.

File: apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts (hunk 3 above)
Comment (rifont) [inline apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts]:
```suggestion

    const workflow = await this.createOrUpdateWorkflow(workflowForUpdate, command);
```

File: apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts [hunk 4]
```diff
@@ -184,6 +184,11 @@ export class UpsertWorkflowUseCase {
     );
   }
 
+  /**
+   * Upsert workflow preferences. While this operation is not typically needed
+   * in a standard workflow update, it's maintained here to support environment
+   * sync scenarios and to enable code reusability.
+   */
```
Comment (rifont) [inline apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts]:
```suggestion
```

File: libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts (hunk 1 above)
Comment (rifont) [inline libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts]: `command.active` is never `null`.

```suggestion
    if (command.active !== undefined) {
```

**Pull request merged**